text/1615-cargo-platform-standards.md

```diff
@@ -0,0 +1,170 @@
+- Feature Name: N/A
+- Start Date: 2016-05-14
+- RFC PR:
+  [rust-lang/rfcs#1615](https://github.com/rust-lang/rfcs/pull/1615)
+- Cargo Issues:
+  [rust-lang/cargo#1734](https://github.com/rust-lang/cargo/issues/1734)
+  [rust-lang/cargo#1976](https://github.com/rust-lang/cargo/issues/1976)
+  [rust-lang/cargo#2127](https://github.com/rust-lang/cargo/pull/2127)
+- Rustup Issues:
+  [rust-lang-nursery/rustup.rs#247](https://github.com/rust-lang-nursery/rustup.rs/issues/247)
+  [rust-lang-nursery/rustup.rs#473](https://github.com/rust-lang-nursery/rustup.rs/issues/473)
+
+# Summary
+
+Improve Cargo's integration into the host operating system by using
+platform-specific paths for config files, cache files and executables,
+so it interacts well with other tools on that platform.
+
+# Motivation
+
+Currently, Cargo puts all its files in a directory named `.cargo` below the
+home directory of the user. Using this proposal, it would reuse existing
+standard directories of the host platform. This allows other tools oblivious to
+Cargo to function in the best way.
+
+Benefits include:
+
+* Using a `.cargo` directory is violating the recommendations/rules on most
+  operating systems. _(Linux, Windows, macOS. Especially painful on Windows,
+  where dotfiles are not hidden.)_
```

Comment:
The hidden attribute is user settable: rustup and cargo are entirely able to set it when creating .cargo and .rustup, making this 'Especially painful' comment a small matter of a bugfix in rustup. https://github.com/rust-lang/rustup/issues/2418

Reply:
I have hidden files to always be displayed, so whether the folder is hidden or not is irrelevant. In fact, I want it to *not* be hidden, because any application that is installed normally is not hidden, so why does rustup/cargo need to hide where it is installed?

Reply:
I too have hidden always shown; I'm mainly responding to the 'especially painful on Windows' narrative in this document, which seems to expect Windows to have the same norms as Linux, but Windows users often don't have the same norms or expectations.

Reply:
Well it *is* especially painful on Windows, not because it isn't hidden, but because it's a dotfile on Windows! It's a blatant sign of software that was written for linux and poorly ported over to Windows, and it leaves a very bad impression of Rust as not supporting Windows that well.

Reply:
I agree with you on that, but that sentence gives an entirely different context :).

Reply:
Sounds like

> , where dotfiles are not hidden

should be removed, then.

Reply:
@retep998 would "Especially painful on Windows, where the use of a dot-prefixed directories and unusual locations gives a strong impression of poor platform support." be a good replacement?

Reply:
Sure.